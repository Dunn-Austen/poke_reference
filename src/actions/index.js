export const cacheNames = pokeNames => ({
  type: 'CACHE_NAMES',
  pokeNames
});

export const cacheTypes = pokeTypes => ({
  type: 'CACHE_TYPES',
  pokeTypes
});

export const handleError = errorMessage => ({
  type: 'HANDLE_ERROR',
  errorMessage
})

export const isLoading = loadingStatus => ({
  type: 'IS_LOADING',
  loadingStatus
})

export const storePokemon = pokeData => ({
  type: 'STORE_POKEMON',
  pokeData
})

export const storeOpponentTypes = opponentTypes => ({
  type: 'STORE_OPPONENTS',
  opponentTypes
})
