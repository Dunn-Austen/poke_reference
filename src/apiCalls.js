export const fetchEmAll = () => {
  return fetch('https://pokeapi.co/api/v2/pokemon/?limit=4000')
    .then(response => {
      if (!response.ok) {
        throw Error('Error with catchEmAll get attempt')
      }
      return response.json()
    })
}

export const fetchTypes = () => {
  return fetch('https://pokeapi.co/api/v2/type')
    .then(response => {
      if (!response.ok) {
        throw Error('Error with fetchTypes')
      }
      return response.json()
    })
}
