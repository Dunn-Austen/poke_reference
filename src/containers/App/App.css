main {
  background-color: grey;
  height: 100vh;
  width: 100%;
}

.sections-container {
  display: flex;
  flex-direction: row;
  height: 90vh;
  width: 100%;
}

.main-left {
  align-items: center;
  display: flex;
  flex-direction: column;
  height: 100%;
  justify-content: center;
  width: 40%;
}

.main-right {
  align-items: center;
  display: flex;
  flex-direction: column;
  height: 100%;
  justify-content: space-around;
  width: 60%;
}

.all-pokemon {
  align-items: center;
  display: flex;
  flex-direction: column;
  height: 70%;
  overflow-x: scroll;
}

.pokemon-listing {
  display: block;
}
