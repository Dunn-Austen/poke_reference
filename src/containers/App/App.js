import React, {Component} from 'react';
import './App.css';
import { Route } from 'react-router-dom';
import { fetchEmAll, fetchTypes } from '../../apiCalls'
import { cacheNames, cacheTypes, handleError, isLoading } from '../../actions';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import SearchBar from '../SearchBar/SearchBar'

export class App extends Component {
  componentDidMount() {
    const { cacheNames, cacheTypes, handleError, isLoading } = this.props
    fetchEmAll()
      .then(data => {
        cacheNames(data.results)
      })
      .catch(error => {
        handleError('Error with pokeNames retrieval')
      })
  }

  render = () => {
    return (
      <main>
        <header>Sample text</header>
        <div className='sections-container'>
          <section className='main-left'>
          </section>
          <section className='main-right'>
            <SearchBar />
          </section>
        </div>

      </main>
    )
  }
}

export const mapStateToProps = state => ({
  pokeNames: state.pokeNames
})

export const mapDispatchToProps = dispatch => ({
  cacheNames: pokeNames => dispatch(cacheNames(pokeNames)),
  cacheTypes: pokeTypes => dispatch(cacheTypes(pokeTypes)),
  handleError: errorMessage => dispatch(handleError(errorMessage)),
  isLoading: loadingStatus => dispatch(isLoading(loadingStatus))
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
App.propTypes = {
  pokeNames:PropTypes.array,
  cacheNames: PropTypes.func,
  cacheTypes: PropTypes.func,
  handleError: PropTypes.func,
  isLoading: PropTypes.func,
}
