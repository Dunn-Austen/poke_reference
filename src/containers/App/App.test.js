import React from 'react';
import { shallow } from 'enzyme';
import { App, mapStateToProps, mapDispatchToProps } from './App';
import { cacheNames, cacheTypes, handleError, isLoading, storeOpponentTypes, storePokemon } from '../../actions';
import { fetchEmAll } from '../../apiCalls';
// import { fetchEmAll, fetchTypes, fetchTypeData, fetchPokemonData, fetchOpponentTypeData  } from './../../apiCalls';

jest.mock('../../apiCalls.js');

describe('App', () => {
  let wrapper;
  let mockCacheNames;
  let mockCacheTypes;
  let mockHandleError;
  let mockIsLoading;
  let mockStoreOpponentTypes;
  let mockStorePokemon;
  let mockPokeNames;

  beforeEach(() => {
    fetchEmAll.mockImplementation(() => {
      return Promise.resolve([
          {name: 'bulbasaur', url: 'https://pokeapi.co/api/v2/pokemon/1/'},
          {name: 'charmander', url: 'https://pokeapi.co/api/v2/pokemon/2/'},
          {name: 'squirtle', url: 'https://pokeapi.co/api/v2/pokemon/3/'}
        ])
    });

    mockCacheNames = jest.fn();
    mockCacheTypes = jest.fn();
    mockHandleError = jest.fn();
    mockIsLoading = jest.fn();
    mockStoreOpponentTypes = jest.fn();
    mockStorePokemon = jest.fn();
    mockPokeNames = [
        {name: 'bulbasaur', url: 'https://pokeapi.co/api/v2/pokemon/1/'},
        {name: 'charmander', url: 'https://pokeapi.co/api/v2/pokemon/2/'},
        {name: 'squirtle', url: 'https://pokeapi.co/api/v2/pokemon/3/'}
      ];

    wrapper = shallow(<App
      cacheNames={mockCacheNames}
      cacheTypes={mockCacheTypes}
      handleError={mockHandleError}
      isLoading={mockIsLoading}
      storePokemon={mockStoreOpponentTypes}
      storeOpponentTypes={mockStorePokemon}
      pokeNames={mockPokeNames}
      />)
  });

  it('should match the Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should invoke fetchEmAll on page load', () => {
    expect(fetchEmAll).toHaveBeenCalled()
  });

  describe('mapsStateToProps', () => {
    it('should return only the pertinent information from the redux store', () => {
      const mockState = {
        pokeNames: [{pokemon: 'name', url: 'url'}],
        errorMessage: '',
        isLoading: false,
        pokeTypes: [],
        pokeData: {},
        opponentTypes: [],
      };
      const expected = {
        pokeNames: [{pokemon: 'name', url: 'url'}],
      };
      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);
    });
  });

  describe('mapDispatchToProps', () => {
    it('calls dispatch with a cacheNames action when cacheNames is called',
      () => {
        const mockDispatch = jest.fn();
        const actionToDispatch = cacheNames([{pokemon: 'name', url: 'url'}]);
        const mappedProps = mapDispatchToProps(mockDispatch);

        mappedProps.cacheNames([{pokemon: 'name', url: 'url'}]);

        expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });

    it('calls dispatch with a cacheTypes action when cacheTypes is called',
      () => {
        const mockDispatch = jest.fn();
        const actionToDispatch = cacheTypes([{name: 'flying', url: 'url'}]);
        const mappedProps = mapDispatchToProps(mockDispatch);

        mappedProps.cacheTypes([{name: 'flying', url: 'url'}]);

        expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });

    it('calls dispatch with a handleError action when handleError is called',
      () => {
        const mockDispatch = jest.fn();
        const actionToDispatch = handleError('error');
        const mappedProps = mapDispatchToProps(mockDispatch);

        mappedProps.handleError('error');

        expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });

    it('calls dispatch with an isLoading action when isLoading is called',
      () => {
        const mockDispatch = jest.fn();
        const actionToDispatch = isLoading(false);
        const mappedProps = mapDispatchToProps(mockDispatch);

        mappedProps.isLoading(false);

        expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });

    it('calls dispatch with an storePokemon action when storePokemon is called',
      () => {
        const mockDispatch = jest.fn();
        const actionToDispatch = storePokemon({name: 'pikachu', abilities: []});
        const mappedProps = mapDispatchToProps(mockDispatch);

        mappedProps.storePokemon({name: 'pikachu', abilities: []});

        expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });

    it('calls dispatch with an storeOpponentTypes action when storeOpponentTypes is called',
      () => {
        const mockDispatch = jest.fn();
        const actionToDispatch = storeOpponentTypes([{damage_relations: {}}]);
        const mappedProps = mapDispatchToProps(mockDispatch);

        mappedProps.storeOpponentTypes([{damage_relations: {}}]);

        expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
})
