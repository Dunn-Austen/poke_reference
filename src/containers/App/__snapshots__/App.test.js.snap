// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the Snapshot 1`] = `
<main>
  <Connect(NavBar) />
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      exact={true}
      path="/pokesearch"
      render={[Function]}
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(PokePage)",
          "type": [Function],
        }
      }
      exact={true}
      path="/pokepage"
    />
    <Route
      component={[Function]}
    />
  </Switch>
</main>
`;
